libs/community/langchain_community/tools/azure_ai_services/translate_tool.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import logging
+import os
+from typing import Any, Optional
+
+from azure.ai.translation.text import TextTranslationClient
+from azure.core.credentials import AzureKeyCredential
+from langchain_core.callbacks import CallbackManagerForToolRun
+from langchain_core.tools import BaseTool
+
+logger = logging.getLogger(__name__)
+
+
+class AzureTranslateTool(BaseTool):
+    """
+    A tool that interacts with the Azure Translator API using the SDK.
+
+    This tool queries the Azure Translator API to translate text between languages.
+    It requires an API key and endpoint, which can be set up as described in the
+    Azure Translator API documentation:
+    https://learn.microsoft.com/en-us/azure/ai-services/translator/
+    translator-text-apis?tabs=python
+    """
+
+    translate_key: str = ""
+    translate_endpoint: str = ""
+    translate_client: Any = None
+    default_language: str = "en"
+
+    name: str = "azure_translator_tool"
+    description: str = (
+        "A wrapper around Azure Translator API. Useful for translating text between "
+        "languages. Input must be text (str). Ensure to install the "
+        "azure-ai-translation-text package."
+    )
+
+    @classmethod
+    def validate_environment(cls):
```

Comment:
I think this should accept the `values` dict like shown here: https://github.com/Sheepsta300/langchain/blob/e5fb3635edc85b7233ebee1ebd58131a9513258d/libs/community/langchain_community/tools/azure_ai_services/image_analysis.py#L39